server/aws-lsp-codewhisperer/src/language-server/chat/contexts/documentContext.test.ts

```diff
@@ -0,0 +1,163 @@
+import { EditorState } from '@amzn/codewhisperer-streaming'
+import * as assert from 'assert'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { DocumentContextExtractor } from './documentContext'
+
+describe('DocumentContext', () => {
+    const mockTypescriptCodeBlock = `function test() {
+    console.log('test')
+}`
+    const mockTSDocument = TextDocument.create('file://test.ts', 'typescript', 1, mockTypescriptCodeBlock)
+
+    describe('documentContextExtractor.extractEditorState', () => {
+        it('extracts editor state for range selection', async () => {
+            const documentContextExtractor = new DocumentContextExtractor(19)
```

Comment:
Does the limit always respect the nearest line boundaries or does it just go `N/2` characters back and `N/2` characters forward?

Reply:
No, this doesn't respect line boundaries, it will go all the way until the characters limit or document boundaries, whichever comes first. If we see an issue with sending partial lines, I can modify the logic.

Reply:
Nope, sounds good to me!

There could be some theoretical edge cases where cutting off partial statements could result in the intent changing

`mock.deleteAllFiles()` vs `deleteAllFiles()`

but lines don't resolve that, only a context-aware parser will be able to catch that. Since the user is already selecting the relevant file contents, and we _increase_ the context that we send, this seems like a highly unlikely edge case.